OverlayPlugin.Core/MemoryProcessors/Combatant/LineCombatant.cs

```diff
@@ -0,0 +1,411 @@
+                ffxiv.RegisterNetworkParser(MessageReceived);
+            }
+            catch (System.IO.FileNotFoundException)
+            {
+                logger.Log(LogLevel.Error, Resources.NetworkParserNoFfxiv);
+            }
+            catch (Exception e)
+            {
+                logger.Log(LogLevel.Error, Resources.NetworkParserInitException, e);
+            }
+
+            cancellationToken = new CancellationTokenSource();
+
+            Task.Run(PollCombatants, cancellationToken.Token);
+        }
+
+        ~LineCombatant()
+        {
+            if (cancellationToken != null)
+            {
+                cancellationToken.Cancel();
+            }
+        }
+
+        private void PollCombatants()
+        {
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                try
+                {
+                    var now = DateTime.Now;
+
+                    CheckCombatants(now);
+
+                    // Wait for next poll
+                    var delay = CombatantChangeCriteria.PollingRate - (int)Math.Ceiling((DateTime.Now - now).TotalMilliseconds);
+                    if (delay > 0)
+                    {
+                        Thread.Sleep(delay);
+                    }
+                    else
+                    {
+                        // If we're lagging enough to not have a sleep duration, delay by PollingRate to reduce lag
+                        Thread.Sleep(CombatantChangeCriteria.PollingRate);
+                    }
+                }
+                catch (Exception e)
+                {
+                    logger.Log(LogLevel.Debug, $"LineCombatant: Exception: {e}");
+                }
+            }
+        }
+
+        private void CheckCombatants(DateTime now, params uint[] filter)
+        {
+            bool inCombat = inCombatMemory.GetInCombat();
+            var combatants = combatantMemoryManager.GetCombatantList();
+
+            // Check combatants currently in memory first
+            foreach (var combatant in combatants)
+            {
+                // If we're only checking specific actor IDs, filter to those
+                if (filter.Length > 0 && !filter.Contains(combatant.ID))
+                {
+                    continue;
+                }
+
+                // If this is a new combatant, always write a line for it
+                if (!combatantStateMap.ContainsKey(combatant.ID))
+                {
+                    combatantStateMap[combatant.ID] = new CombatantStateInfo()
+                    {
+                        lastUpdated = now,
+                        combatant = combatant,
+                    };
+                    WriteLine(
+                        CombatantMemoryChangeType.Add,
+                        combatant.ID,
+                        string.Join("", CombatantChangeCriteria.AllFields.Select((fi) => FormatFieldChange(fi, combatant, true))));
```

Comment:
It is a little surprising to me that `Add` lists a lot of fields that `Change` does not.  For example, `CurrentHP` or `RawEffectiveDistance`.  Do you think it would make sense to have a single set of fields that Add/Change care about for consistency?

Reply:
`Change` *should* be able to list all the same fields that `Add` can. Looks like I missed a loop to add other changed props when a prop was changed.

Reply:
That said, probably be we should skip `CurrentHP` and `RawEffectiveDistance`? That seems like it would add quite a few lines and is already captured in other information.

Reply:
I looked through your example logs and didn't see any `CurrentHP`, but maybe I made a mistake?

Not including it at all seems good too.

Reply:
It was excluded from `Change` lines previously due to the aforementioned missing loop, but was intended to be included.

The intended behavior here is:
1. Check any shorter-than-default values (`BNpcNameID` [0ms], `TransformationId` [0ms], `WeaponId` [0ms], `TargetID` [0ms], `Pos*` [250ms], `Heading` [250ms]). If any of these values have changed and their associated delay has passed, queue up a `Change` line
   - `Pos*` and `Heading` are treated specially to force always including `PosX`, `PosY`, `PosZ`, and `Heading` if either of the thresholds have been passed
2. Check all other properties defined by `DefaultCheckFields` (`OwnerID`, `Type`, `MonsterType`, `Status`, `ModelStatus`, `AggressionStatus`, `IsTargetable`, `Name`, `Radius`, `BNpcID`, `CurrentMP`, and `IsCasting1`). If any of them have changed, and the last time a change line was printed exceeds the default threshold (1000ms), queue up a `Change` line
3. If a `Change` line is queued, check every changed property on the combatant and include the new value in the line, based on fields defined in `AllFields`. This excludes fields defined in `IngoreFieldNames` (`ID`, `Effects`, `RawEffectiveDistance`, and `CurrentHP`)

As a full chart, the way this works currently on a per-field basis is:

Field | Included if changed | Default Checked (1000ms) | Specific check delay | Has other threshold | Notes
-- | -- | -- | -- | -- | --
ID | No | No | No | No | Included in every line (except Clear)
OwnerID | Yes | Yes | No | No |  
Type | Yes | Yes | No | No |  
MonsterType | Yes | Yes | No | No |  
Status | Yes | Yes | No | No |  
ModelStatus | Yes | Yes | No | No |  
AggressionStatus | Yes | Yes | No | No |  
TargetID | Yes | No | 0ms | No |  
IsTargetable | Yes | Yes | No | No |  
Job | Yes | No | No | No |  
Name | Yes | Yes | No | No |  
CurrentHP | No | No | No | No |  
MaxHP | Yes | No | No | No |  
PosX | Yes | No | 250ms | Yes |  
PosY | Yes | No | 250ms | Yes |  
PosZ | Yes | No | 250ms | Yes |  
Heading | Yes | No | 250ms | Yes |  
Radius | Yes | Yes | No | No |  
Distance | Yes | No | No | No | Will currently never have data
EffectiveDistance | Yes | No | No | No | Will currently never have data
RawEffectiveDistance | No | No | No | No |  
Effects | No | No | No | No | Excluded because of complexity
BNpcID | Yes | Yes | No | No |  
CurrentMP | Yes | Yes | No | No |  
MaxMP | Yes | No | No | No |  
Level | Yes | No | No | No |  
BNpcNameID | Yes | No | 0ms | No |  
WorldID | Yes | No | No | No |  
CurrentWorldID | Yes | No | No | No |  
NPCTargetID | Yes | No | No | No |  
CurrentGP | Yes | No | No | No |  
MaxGP | Yes | No | No | No |  
CurrentCP | Yes | No | No | No |  
MaxCP | Yes | No | No | No |  
PCTargetID | Yes | No | No | No |  
IsCasting1 | Yes | Yes | No | No |  
IsCasting2 | Yes | No | No | No |  
CastBuffID | Yes | No | No | No |  
CastTargetID | Yes | No | No | No |  
CastDurationCurrent | Yes | No | No | No |  
CastDurationMax | Yes | No | No | No |  
TransformationId | Yes | No | 0ms | No |  
WeaponId | Yes | No | 0ms | No |  

From a data integrity standpoint I think it makes more sense to not exclude any raw-from-memory values from the `Add` and `Change` lines. In fact, I'd like to include `Effects` as well but it's not so easy to serialize that down to a flat line format.

Also, despite the `Common.cs` `Combatant` struct having `Distance`, `EffectiveDistance`, and `RawEffectiveDistance`, only the latter is ever actually populated for this case. The two former values are only populated by `EnmityEventSource.cs` and should probably be moved upstream to the `CombatantMemory` class structure somewhere in a separate PR.

Reply:
> I'd like to include Effects as well but it's not so easy to serialize that down to a flat line format.

I'm not sure if effects would be necessary, since it's the same data that 38-lines already have.